pilot/pkg/xds/endpoint_builder.go

```diff
@@ -74,14 +77,15 @@ func NewEndpointBuilder(clusterName string, proxy *model.Proxy, push *model.Push
 	_, subsetName, hostname, port := model.ParseSubsetKey(clusterName)
 	svc := push.ServiceForHostname(proxy, hostname)
 	return EndpointBuilder{
-		clusterName:     clusterName,
-		network:         proxy.Metadata.Network,
-		networkView:     model.GetNetworkView(proxy),
-		clusterID:       proxy.Metadata.ClusterID,
-		locality:        proxy.Locality,
-		service:         svc,
-		destinationRule: push.DestinationRule(proxy, svc),
-		tunnelType:      GetTunnelBuilderType(clusterName, proxy, push),
+		clusterName:         clusterName,
+		network:             proxy.Metadata.Network,
+		networkView:         model.GetNetworkView(proxy),
+		clusterID:           proxy.Metadata.ClusterID,
+		locality:            proxy.Locality,
+		service:             svc,
+		destinationRule:     push.DestinationRule(proxy, svc),
+		peerAuthentications: push.AuthnPolicies.GetPeerAuthenticationsForNamespace(proxy.ConfigNamespace),
```

Comment:
Can you run this with `UNSAFE_PILOT_ENABLE_RUNTIME_ASSERTIONS`? When doing the CDS cache I ran into some issues that *might* impact this but I forget the details- I will go refresh my memory and come back to this

Reply:
Actually I think the existing EDS cache has it, with and without this PR. I will need to verify, its late so will do it tomorrow.

My concern is:
* We start with push == v1, and 2 goroutines generating EDS (g1 and g2)
* g1 builds key, gets DRv1
* DR updated to v2, cache invalidated
* g1 builds EDS and writes it
* g2 builds key, gets DRv2. Version is not part of the key, so we have a cache hit, and return the EDS for the old DR

In the CDS PR (https://github.com/istio/istio/pull/31193/files) I added `pushVersion: cb.push.PushVersion` to the key, maybe that will help here.

Its not really made worse by this PR so doesn't need to block

Reply:
Yeah it seems like things are definitely not being evicted when they should. I'm not 100% sure if it's due to the PeerAuthentication/dependent configs logic or if I'm hitting the issue you're describing. For now I'm copying in the `DependentTypes` change.